charts/seed-controlplane/charts/etcd/templates/etcd.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: druid.gardener.cloud/v1alpha1
+kind: Etcd
+metadata:
+  name: etcd-{{ .Values.role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    garden.sapcloud.io/role: controlplane
+    role: {{ .Values.role }}
+{{- if .Values.labels }}
+{{ toYaml .Values.labels | indent 4 }}
+{{- end }}
+spec:
+{{- if .Values.podAnnotations }}
+  annotations:
+{{ toYaml .Values.podAnnotations | indent 4 }}
+{{- end }}
+  labels:
+    garden.sapcloud.io/role: controlplane
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+    networking.gardener.cloud/to-dns: allowed
+    networking.gardener.cloud/to-public-networks: allowed
+    networking.gardener.cloud/to-private-networks: allowed
+  etcd:
+    tls:
+      serverTLSSecretRef:
+        name: {{ .Values.tlsServerSecretName }}
+        namespace: {{ .Release.Namespace }}
+      clientTLSSecretRef:
+        name: {{ .Values.tlsClientSecretName }}
+        namespace: {{ .Release.Namespace }}
+    metrics: {{ .Values.metrics }}
+    version: {{ .Values.etcdVersion }}
```

Comment:
This is only the image tag, but where is the image repository specified?

Reply:
I have kept the repository to be not configurable. Only the tags can be updated. The repository information is fetched from [here](https://github.com/gardener/etcd-druid/blob/master/charts/images.yaml).

Reply:
I think we should make the repository configurable, especially for the use-case where Gardener is deployed in private clouds or on-prem. In these cases it is likely that the container images are copied into a local registry and should be downloaded from there (instead of from the public registries hard-coded in the file you linked).

Reply:
Ah ok. I'll change the code to include repository as well.

Reply:
Currently [here](https://github.com/gardener/etcd-druid/blob/master/controllers/etcd_controller.go#L98), there is an option to pull artifactory urls and images from the `images.yaml` file. If I add a command-line parameter to druid, and use that as the path to this file to pull the image url and version, would that work as well?

Reply:
It would work, sure, but it would be more complex. It's safer to just expose the full image in the `Etcd` resource and not only the tag. This way only Gardener needs to be configured with potential repository overwrites and etcd-druid can be kept untouched.

Reply:
I don't think this needs any change. The image repository is already configurable at global leve in etcd-druid. Etcd-druid uses the imagevector lib from gardener to load image info [Ref](https://github.com/gardener/etcd-druid/blob/master/controllers/etcd_controller.go#L670). So, by passing customised image vector path to etcd-druid via environment variable will do the job. I don't think image repository should be part of etcd CRD and it should be only configurable at etcd-druid level globally.

Reply:
I disagree. It's a) more complex to overwrite the image vector in etcd-druid (because you first need to configure Gardener with a dedicated image vector overwrite for the druid, and then pass it through to the druid chart during seed bootstrap), and b) you should always be able to specify the location if you can specify the version. But the complexity is the main argument, obviously.

Reply:
As you wish i'm fine with both of them. Said that, i don't find complexity varying much in both cases. IMHO from just etcd-druid as standalone PoV option [A] makes more sense to me.

Reply:
Yes, as stand-alone component it makes sense, I agree. Can we settle on making the repository configurable as well, but optional? If not provided, etcd-druid can take its default repo, but for end-users like Gardener it's possible to overwrite it without the necessity to re-configure the druid itself. WDYT? This would also make use-cases possible where you want to use the standard image from etcd-druid for `Etcd/first-resource` and another image for `Etcd/second-resource`, i.e., for multiple `Etcd` resources.

Reply:
Yeah. As i said i'm fine with it. No strong opinion against it.